src/routes/theatre.js

```diff
@@ -0,0 +1,70 @@
+const express = require("express");
+const Theatre = require("../models/theatres");
+const MovieTheatreMapping = require("../models/movieTheatreMappings");
+const router = express.Router();
+
+//get All theatres or based on city
+router.get("/", async (req, res) => {
+  try {
+    const { city } = req.query;
+    let theatres = [];
+    if (city) {
+      theatres = await Theatre.find({
+        where: {
+          city: city,
+        },
+      });
+    } else {
+      theatres = await Theatre.findAll();
+    }
+    if (theatres.length == 0) {
+      return res.status(404).json({ message: "No Result Found." });
+    }
+    return res.status(200).json(theatres);
+  } catch (error) {
+    res.status(500).json({ message: "Something Went Wrong!" });
+  }
+});
+
+router.get("/:theatreId", async (req, res) => {
+  try {
+    const { theatreId } = req.params;
+    const { date } = req.query;
+    let theatreMovies = await MovieTheatreMapping.findAll({
+      where: {
+        theatreId: theatreId,
+        date: new Date(date),
```

Comment:
might give wrong result in case of incorrect value for date. For example I hit the request with date='random'. Date constructor will parse this to some random value. A util function to check if the date is valid before making a DB query would have been better.